sdk/core/Azure.Core/src/Shared/JsonPropertyExtensions.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.Json;
+
+namespace Azure.Core
```

Comment:
I'll defer to @pakrym on whether these should go in Core.  @AlexanderSher, have you looked at unit tests across libraries to see if other tests would make use of this?  I'm also wondering ... as we move forward to autogenerated models and serialization, would it make sense to have these extensions?  If autogeneration doesn't use them, then they'll only be there for hand-written serialization, and that might be a little odd re: consistency.  It might be worth starting a thread with the .NET team to discuss pros and cons here.

Reply:
I would prefer not to add these extension method to Azure.Core shared right now for the following reasons:
2. While these are in shared directory they are only used by app config. I'm reluctant to add anything to shared until it's truly shared.
1. We are planning to add a bunch of extension method as part of codegen work and I'm afraid these would not get enough use.